### DISCUSSION TITLE
TOPIC creation

### DISCUSSION THREAD

**Discussion by sreejesh-radhakrishnan-db** [strimzi/strimzi-kafka-operator#7018] [General]
So I wanted to know what qualify as an improvement ,I spend almost all day today debugging why TOPIC are not becoming READY state

No error in entity operator or strimzi operator or any LOG in GKE which can help me with debug. 

Issue was due to PICNIC , I admit

the CLUSTER name I gave was wrong and that was the root cause for TOPIC not bring in READY state 

{{ .Values.kafka_cluster_name }}  =  gpds-uat  is the name I should have given instead I parameterized it as 
{{ .Values.kafka_cluster_name }}  =  gpds-uat-kafka  (error as kafka cluster append a -kafka to it which caused me this error)

My Q - when cluster name is NOT correct  and a kafka cluster with that name never exist in my K8s, why is OPERATOR creating the TOPIC with READY status EMPTY?   wont it be ideal if they throw error stating the obvious? 

My TOPIC file:

apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaTopic
metadata:
                   name: test.
  labels:
                      strimzi.io/cluster: {{ .Values.kafka_cluster_name }}
spec:
                partitions: 3
                replicas: 3
  config:
                  retention.ms: 7200000
                 segment.bytes: 1073741824

**Comment by scholzj** [comment]
The operator watches the topic based on the label. So when it has a wrong label, the operator does not see it and cannot do anything about it. 

[This blog](https://strimzi.io/blog/2020/09/01/enforce-custom-resource-policies-with-opa-gatekeeper/) post describes how you can use OPA Gatekeeper to protect against this (the blog post is older, so you will probably need to modify things a bit - update the APIs etc.).
